### PR TITLE
[Term Entry] Neural Networks ai RMSprop

### LINKED ISSUE #4529: [Term Entry] Neural Networks ai RMSprop
### Reason/inspiration (optional)

"We would like a new term entry in the `AI` concept for [neural-network](https://www.codecademy.com/resources/docs/ai/neural-networks): RMSprop. The entry should go in a new file under `docs/content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md`.

The entry should describe the term and how it is used. Also include: 
- An "Example" section that explains to readers how to use concept for stochastic gradient descent.
- Should contain information for this concept and syntax.
- Should have a Python codebyte with a working code snippet showcasing this concept.
Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added new term entry for AI: Neural Networks - RMSprop

### Issue Solved

Closes #4529 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+**RMSprop** is an abbreviation for the term, **Root Mean Square Propagation**, and is an algorithm that is designed to adjust the learning rates dynamically for training neural network models. It helps to improve the stability and speed of the training process by adapting the learning rate based on recent gradients, making it effective for training deep neural networks.
+
+## How Does RMSprop Work?
+
+RMSprop modifies the learning rate for each parameter individually by taking into account the magnitudes of the recent gradients to adjust the updates. Here is how it works:
+
+- **Exponential Decay of Squared Gradients**: RMSprop keeps track of the moving average of the squared gradients of the loss function with an exponential decay factor. The average is then used to normalise the gradients, preventing them from being too large or too small.
+- **Update Rule**: For each parameter $\theta_i$, RMSprop updates the exponentially decaying average of the squared gradients $E[g^2]_t$ and then uses it to adjust the parameter:
+  $$E[g^2]_t = \rho E[g^2]_{t-1} + (1-\rho)g^2_t$$
+  $$\theta_{t+1} = \theta_t - \frac{\eta}{\sqrt{E[g^2]_t + \epsilon}}g_t$$
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md:27]: Can this be represented either in an image or in a python equation format? Since this is a LaTeX equation, we need to present this in a different format.

File: content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  - 'Computer Science'
+Tags:
+  - 'AI'
+  - 'Machine Learning'
+  - 'Neural Networks'
+  - 'Functions'
+CatalogContent:
+  - 'paths/machine-learning-ai-engineering-foundations'
+  - 'intro-to-ai-strategy'
+---
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md:16]:
```suggestion
---
Title: 'RMSprop'
Description: 'RMSprop is an optimization algorithm that dynamically adjusts the learning rate during the training of neural networks.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Machine Learning'
Tags:
  - 'AI'
  - 'Functions'
  - 'Machine Learning'
  - 'Neural Networks'
CatalogContent:
  - 'paths/machine-learning-ai-engineering-foundations'
  - 'intro-to-ai-strategy'
---
```

Comment (mamtawardhani) [reply content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md:16]: Arranged the tags and subjects in an alphabetical manner. Reframed the description to be more readable.

File: content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+# Example gradients from a loss function
+gradients = np.array([-0.5, 0.3, 0.24])
+
+# Update running average
+E_g2 = rho * E_g2 + (1 - rho) * gradients**2
+
+# Update parameters using RMSprop
+theta -= eta / np.sqrt(E_g2 + epsilon) * gradients
+
+print("Updated parameters:", theta)
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md:57]:
```suggestion
import numpy as np

# Example neural network parameters (weights)
theta = np.array([0.2, 0.4, 0.6])

# Learning rate (step size for updating parameters)
eta = 0.01

# Decay rate (used to compute the moving average of squared gradients)
rho = 0.9

# Small constant to prevent division by zero
epsilon = 1e-8

# Running average of squared gradients
E_g2 = np.zeros_like(theta)

# Example gradients of the loss function with respect to the parameters
gradients = np.array([-0.5, 0.3, 0.24])

# Update running average of squared gradients
E_g2 = rho * E_g2 + (1 - rho) * gradients**2

# Update parameters using RMSprop
theta -= eta / np.sqrt(E_g2 + epsilon) * gradients

print("Updated parameters:", theta)
```

Comment (mamtawardhani) [reply content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md:57]: Reframed the comments for better understanding.

File: content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+The output of the above code is as follows:
+
+```shell
+Updated parameters: [0.23162277 0.36837724 0.56837725]
+```
+
+## Codebyte Example
+
+The following Codebyte example also demonstrates the use of RMSprop in another stochastic gradient descent scenario:
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/rmsprop/rmsprop.md:68]:
```suggestion
Run the following Codebyte to understand the use of RMSprop in a stochastic gradient descent scenario:
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @andersooi, firstly apologies for the delay in the review.  😢 
Thank you for contributing to the Codecademy Docs, the entry is well written! 😄 

I've suggested a few modifications, could you please review and modify them at your earliest convenience? Thank you! 😃

**Review by mamtawardhani — APPROVED** [review]
Hey @andersooi thank you for contributing! 

the entry looks good for a second round of review! 😄

**Comment by mamtawardhani** [comment]
Hey @Sriparno08, will you be able to review this? This entry is from two months ago actually 🫨

**Comment by Sriparno08** [comment]
Yeah sure, @mamtawardhani!

**Review by Sriparno08 — APPROVED** [review]
Looks good, @andersooi! 👍
